### DISCUSSION TITLE
Device Access for Developers at Remote Offices

### DISCUSSION THREAD

**Discussion by ryanzor** [MobileNativeFoundation/discussions#176] [General]
Hi all.  I'm at Pinterest and we are re-examining our strategy to get devices to developers.  Before remote work the office would have a variety of devices developers could check out as needed.  When we went remote the core strategy has been to send devices to developers when they need them and let them keep them as long as needed.  This has run into the problem that it is sometimes hard for developers to anticipate they'll need a device and international shipping has been extremely slow.

We are looking at remote solutions such as [BrowserStack AppLive](https://www.browserstack.com/app-live) that naturally will scale better for our needs.  I'm generally curious how other companies are solving this.

**Comment by fbaptista** [comment]
Hello @ryanzor , same curiosity here! 
Working at Apptim we've seen lot of companies using DeviceFarmer (ex OpenSTF) for Android
https://github.com/DeviceFarmer/stf

unfortunately, iOS support seems suspended and some forks efforts https://github.com/DeviceFarmer/stf_ios_support/ are still very hard to use.

**Reply by donnie-jp** [reply]
It was created primarily for security testing but the ARM native (not simulator/emulator) https://www.corellium.com/solutions/mobile-app-testing could be an option. It has limitations in terms of sensor support but should deal with most use cases.

I suggested it for our security pentesters.

**Comment by kapoorlakshya** [comment]
At Reddit, we're using BrowserStack App Live and it has been working well so far. We do occasionally run into a situation where we want to reproduce a user reported bug on a device+OS combo that's not in this [device list](https://www.browserstack.com/list-of-browsers-and-platforms/app_live), but not a deal breaker.
